File: Challenge 3/challenge3.py
class OrderProcessor:
    def process_order(self, customer_name, items, apply_discount):
        total_price = 0

        for item in items:
            if item == "ItemA":
                total_price += 100
            elif item == "ItemB":
                total_price += 200
            elif item == "ItemC":
                total_price += 300
            else:
                total_price += 50

        if apply_discount:
            total_price *= 0.9 

        print(f"Order for {customer_name}:")
        for item in items:
            print(f"- {item}")
        print(f"Total price: {total_price}")

        self.send_email_receipt(customer_name, total_price)

    def send_email_receipt(self, customer_name, total_price):
        print(f"Sending email to {customer_name} with total price of {total_price}")


if __name__ == "__main__":
    processor = OrderProcessor()

    items = ["ItemA", "ItemB", "ItemX"] 
    processor.process_order("John Doe", items, True)

File: Challenge 3/challenge3.cs
using System;
using System.Collections.Generic;

class OrderProcessor {
    public void ProcessOrder(string customerName, List<string> items, bool applyDiscount) {
        double totalPrice = 0;

        foreach (string item in items) {
            if (item == "ItemA")
                totalPrice += 100;
            else if (item == "ItemB")
                totalPrice += 200;
            else if (item == "ItemC")
                totalPrice += 300;
            else
                totalPrice += 50;
        }

        if (applyDiscount) {
            totalPrice *= 0.9;  
        }

        Console.WriteLine($"Order for {customerName}:");
        foreach (string item in items) {
            Console.WriteLine($"- {item}");
        }

        Console.WriteLine($"Total price: {totalPrice}");

        SendEmailReceipt(customerName, totalPrice);
    }


    private void SendEmailReceipt(string customerName, double totalPrice) {
        Console.WriteLine($"Sending email to {customerName} with total price of {totalPrice}");
    }
}

class Program {
    static void Main(string[] args) {
        OrderProcessor processor = new OrderProcessor();

        List<string> items = new List<string> { "ItemA", "ItemB", "ItemX" }; 
        processor.ProcessOrder("John Doe", items, true);
    }
}